develop/parachain-devs/runtime-development/FRAME/create-custom-pallet.md

```diff
@@ -0,0 +1,334 @@
+
+        Proper version management is crucial for ensuring compatibility and reducing potential conflicts in your project.
+
+3.  Set up the basic pallet structure
+
+    After setting up the project and configuring dependencies, the next step is to prepare the basic structure of your pallet in the `lib.rs` file. Open the `src/lib.rs` file in your project, remove all existing content and copy and paste the following scaffold code:
+
+    ```rust
+    pub use pallet::*;
+
+    #[frame_support::pallet]
+    pub mod pallet {
+        use frame_support::pallet_prelude::*;
+        use frame_system::pallet_prelude::*;
+
+        #[pallet::pallet]
+        pub struct Pallet<T>(_);
+
+        #[pallet::config]  // snip
+        #[pallet::event]   // snip
+        #[pallet::error]   // snip
+        #[pallet::storage] // snip
+        #[pallet::call]    // snip
+    }
+    ```
+
+    With this scaffold in place, you're ready to start implementing the specific logic and features of your custom pallet. The subsequent sections of this guide will walk you through populating each of these components with the necessary code for your pallet's functionality.
+
+## Pallet Configuration
+
+Every pallet includes a Rust trait called [`Config`](https://paritytech.github.io/polkadot-sdk/master/frame_system/pallet/trait.Config.html){target=\_blank}, which is used to expose configurable options and link your pallet to other parts of the runtime. All types and constants that the pallet depends on must be declared within this trait. These types are defined generically and made concrete when the pallet is instantiated in the `runtime/src/lib.rs` file of your blockchain.
+
+In this step, you'll configure the pallet to emit events. 
+
+Replace the existing #[pallet::config] line with the following code block:
+
+```rust
+#[pallet::config]
+pub trait Config: frame_system::Config {
+    /// The overarching runtime event type.
+    type RuntimeEvent: From<Event<Self>> + IsType<<Self as frame_system::Config>::RuntimeEvent>;
+    /// A type representing the weights required by the dispatchables of this pallet.
+    type WeightInfo: WeightInfo;
+}
+```
+
+## Pallet Events
+
+After configuring the pallet to emit events, the next step is to define the events that can be triggered by functions within the pallet. Events provide a straightforward way to inform external entities such as dApps, chain explorers, or users, that a significant change has occurred in the runtime. In a FRAME pallet, the details of each event and its parameters are included in the node’s metadata, making them accessible to external tools and interfaces.
+
+The `generate_deposit` macro generates a `deposit_event` function on the Pallet, which converts the pallet’s event type into the `RuntimeEvent` (as specified in the `Config` trait) and deposits it using `frame_system::Pallet::deposit_event`.
```

Comment:
Add reference to generate_deposit, deposit_event, RuntimeEvent from the rust docs of the polkadot sdk